sql/api/src/main/scala/org/apache/spark/sql/streaming/StatefulProcessorHandle.scala

```diff
@@ -72,6 +72,26 @@ private[sql] trait StatefulProcessorHandle extends Serializable {
    */
   def getListState[T](stateName: String, valEncoder: Encoder[T]): ListState[T]
 
+  /**
+   * Function to create new or return existing list state variable of given type
+   * with ttl. State values will not be returned past ttlDuration, and will be eventually removed
+   * from the state store. Any state update resets the ttl to current processing time plus
+   * ttlDuration.
+   *
+   * The user must ensure to call this function only within the `init()` method of the
+   * StatefulProcessor.
+   *
+   * @param stateName  - name of the state variable
+   * @param valEncoder - SQL encoder for state variable
+   * @param ttlConfig  - the ttl configuration (time to live duration etc.)
+   * @tparam T - type of state variable
+   * @return - instance of ValueState of type T that can be used to store state persistently
```

Comment:
scaladoc seems to point to ValueState. 

`instance of ListState of type T that can be used to store state persistently`